lib/modules/manager/pipenv/artifacts.ts

```diff
@@ -78,6 +83,42 @@ export function getPipenvConstraint(
   return '';
 }
 
+function getMatchingHostRule(url: string | undefined): HostRule {
+  const scopedMatch = find({ hostType: PypiDatasource.id, url });
+  return is.nonEmptyObject(scopedMatch) ? scopedMatch : find({ url });
+}
```

Comment:
This is taken from Poetry's `artifacts.ts`

Reply:
`find()` always returns a truthy result so I think this fallback logic actually does nothing

Reply:
@viceice @rarkins please let me know if I should remove it from the Pipenv's function. However, I think removing it from Poetry needs to be done in a separate PR.

I also want to point out that IMHO removing it (from pipenv & Poetry) will [contradict with the documentation](https://docs.renovatebot.com/getting-started/private-packages/#dependency-version-lookups), and might break backward compatibility (in case of Poetry) - mind text in bold:

> Configuring Renovate with credentials requires hostRules. Each host rule consists of a hostType value **and/or** a way to match against hosts using matchHost.
> **hostType is not particularly important at this step unless you have different credentials for the same host, but it is sometimes useful in later steps so is good to include if you can.**

Reply:
@rarkins done. please re-review